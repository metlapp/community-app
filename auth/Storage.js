import AsyncStorage from "@react-native-community/async-storage";

const storeUser = async (user) => {
  console.log(user);
  try {
    await AsyncStorage.setItem("userData", JSON.stringify(user));
  } catch (error) {
    console.log("Error storing user", error);
  }
};

const getUser = async () => {
  try {
    let userData = await AsyncStorage.getItem("userData");
    return JSON.parse(userData);
  } catch (error) {
    console.log("Error getting User", error);
  }
};

const removeUser = async () => {
  try {
    await AsyncStorage.removeItem("userData");
  } catch (e) {
    console.log("Error removing User", error);
  }
};

export default { storeUser, getUser, removeUser };
