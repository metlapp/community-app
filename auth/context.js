import React from "react";

export const FormContext = React.createContext();
const AuthContext = React.createContext();

export default AuthContext;
