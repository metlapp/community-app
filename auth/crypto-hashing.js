import * as Crypto from "expo-crypto";

export default async function runCrypto(pass) {
  const digest = await Crypto.digestStringAsync(
    Crypto.CryptoDigestAlgorithm.SHA256,
    pass
  );
  return digest;
}
