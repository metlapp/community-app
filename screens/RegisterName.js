import React, { useContext, useRef, useState } from "react";
import { StyleSheet, View } from "react-native";

import SubmitButton from "../components/SubmitButton";
import AppFormField from "../components/AppFormField";
import Form from "../components/Form";
import AppButton from "../components/AppButton";
import { FormContext } from "../auth/context";

export default function RegisterName() {
  const [name, setName] = useState();
  const { setStep, formData, setFormData } = useContext(FormContext);

  return (
    <View style={styles.container}>
      <Form
        initialValues={{ name: "" }}
        style={styles.form}
        onSubmit={(values) => {
          setFormData(() => ({
            ...formData,
            name: values.name,
          }));
        }}
      >
        <AppFormField
          placeholder="Enter your name"
          name="name"
          style={styles.TextInput}
          setValue={setName}
          width="100%"
        />
        <View>
          <AppButton title="Back" onPress={() => setStep((step) => step - 1)} />
          <SubmitButton title="Save" />
        </View>
      </Form>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    width: "100%",
  },
  form: {
    height: "100%",
    width: "100%",
    backgroundColor: "yellow",
  },
  TextInput: {
    lineHeight: 25,
    fontSize: 20,
    paddingLeft: 10,
  },
});
