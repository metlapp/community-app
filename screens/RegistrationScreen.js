import React, { useState } from "react";
import { SafeAreaView, StyleSheet, Text } from "react-native";
import * as Yup from "yup";

import { FormContext } from "../auth/context";
import RegisterEmail from "./RegisterEmail";
import RegisterName from "./RegisterName";
import RegisterPassword from "./RegisterPassword";

const validationSchema = Yup.object().shape({
  currentPassword: Yup.string().required().min(8).label("Current Password"),
  newPassword: Yup.string().required().min(8).label("New Password"),
  confirmNewPassword: Yup.string().required().min(8).label("Confirm Password"),
});

export default function RegistrationScreen() {
  const [step, setStep] = useState(1);
  const [formData, setFormData] = useState({
    email: "",
    password: "",
    name: "",
  });

  return (
    <FormContext.Provider value={{ formData, setFormData, step, setStep }}>
      <SafeAreaView style={styles.container}>
        <Text style={styles.text}>Registration Page</Text>
        {step === 1 ? (
          <>
            <Text style={styles.text}>Step 1: Enter your email</Text>
            <RegisterEmail setFormData={setFormData} formData={formData} />
          </>
        ) : step === 2 ? (
          <>
            <Text style={styles.text}>Step 2: Enter your password</Text>
            <RegisterPassword setFormData={setFormData} formData={formData} />
          </>
        ) : (
          <>
            <Text style={styles.text}>Step 3: Enter your name</Text>
            <RegisterName setFormData={setFormData} formData={formData} />
          </>
        )}
        {console.log(formData)}
      </SafeAreaView>
    </FormContext.Provider>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
  },
  text: {
    fontSize: 24,
    fontWeight: "600",
    marginBottom: 30,
    marginTop: 5,
  },
});
