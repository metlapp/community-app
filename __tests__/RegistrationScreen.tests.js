import "react-native";
import React from "react";
import { mount, shallow } from "enzyme";
import { act, fireEvent, render, waitFor } from "@testing-library/react-native";
import RegistrationScreen from "../screens/RegistrationScreen";
import RegisterName from "../screens/RegisterName";
import { FormContext } from "../auth/context";

// Note: this is just for use with Jest snapshot testing
// and comes packaged with react-native init project.
// You do not need this if using Enzyme 'toMatchSnapshot' etc.

import renderer from "react-test-renderer";

describe("Testing Registraion Page", () => {
  it("renders correctly, test using Jest", () => {
    renderer.create(<RegistrationScreen />);
  });

  it("matches the snapshot", () => {
    const tree = renderer.create(<RegistrationScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("text on change password screen", () => {
    const wrapper = shallow(<RegistrationScreen />);
    const text = wrapper.find("Text").first().render().text();
    expect(text).toEqual("Registration Page");
  });
});

it("should save email to user object", async () => {
  const formData = jest.fn();
  const setFormData = jest.fn();
  const step = jest.fn();
  const setStep = jest.fn();
  const { getByText, getByTestId, getByPlaceholderText, debug } = render(
    <FormContext.Provider value={{ formData, setFormData, step, setStep }}>
      <RegistrationScreen />
    </FormContext.Provider>
  );

  const title = getByTestId("title");
  const emailTitle = getByTestId("emailTitle");
  const emailInput = getByTestId("emailInput");
  const emailNextButton = getByText("Next");

  expect(title).toBeDefined();
  expect(emailTitle).toBeDefined();
  expect(emailInput).toBeDefined();
  expect(emailNextButton).toBeDefined();

  fireEvent(emailInput, "onChangeText", "this@email.com");

  act(() => {
    fireEvent(emailNextButton, "onPress");
  });

  console.log(debug());
});

// it("Should render finished screen after form submission", () => {
//   const formData = jest.fn();
//   const setFormData = jest.fn();
//   const step = jest.fn();
//   const setStep = jest.fn();

//   const { getByText, getByTestId } = render(
//     <FormContext.Provider value={{ formData, setFormData, step, setStep }}>
//       {setFormData({
//         email: "jerry@gmail.com",
//         password: "jerrypassword",
//         name: "Jerry",
//       })}
//       <RegisterName />
//     </FormContext.Provider>
//   );

//   fireEvent(getByTestId("input"), "onChangeText", "john");

//   //   const errorMessage = wrapper.find(".errorMessage");

//   fireEvent(getByText("Save").parent, "onPress");

//   const text = getByText("Registration Successful!");

//   expect(text).toEqual("Registration Successful!");
// });

//   it("should call handleSubmit after submit", async () => {
//     const handlePasswordChange = jest.fn();
//     const setUser = jest.fn();

//     const { getByText, getByPlaceholderText } = render(
//       <ChangePasswordScreen
//         handlePasswordChange={handlePasswordChange}
//         setUser={setUser}
//       />
//     );

//     const currentPassword = getByPlaceholderText("Current Password");
//     const newPassword = getByPlaceholderText("New Password");
//     const confirmPassword = getByPlaceholderText("Confirm Password");

//     act(() => {
//       fireEvent(currentPassword, "onChangeText", "password");
//     });
//     act(() => {
//       fireEvent(newPassword, "onChangeText", "Peyton12");
//     });
//     act(() => {
//       fireEvent(confirmPassword, "onChangeText", "Peyton12");
//     });

//     act(() => {
//       fireEvent(getByText("Save").parent, "press");
//     });

//     waitFor(() => {
//       expect(handlePasswordChange).toHaveBeenCalledTimes(1);
//     });
//     waitFor(() => {
//       expect(setUser).toHaveBeenCalledTimes(1);
//     });
//   });
// });
