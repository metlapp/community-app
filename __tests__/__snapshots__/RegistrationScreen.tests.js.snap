// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Registraion Page matches the snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "600",
        "marginBottom": 30,
        "marginTop": 5,
      }
    }
    testID="title"
  >
    Registration Page
  </Text>
  <Text
    style={
      Object {
        "fontSize": 24,
        "fontWeight": "600",
        "marginBottom": 30,
        "marginTop": 5,
      }
    }
    testID="emailTitle"
  >
    Step 1: Enter your email
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "height": 50,
          "width": "100%",
        }
      }
    />
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#f2f2f2",
            "borderRadius": 25,
            "height": 55,
            "justifyContent": "center",
            "marginVertical": 15,
          },
          Object {
            "width": "100%",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "rgb(231, 231, 231)",
              "borderTopLeftRadius": 4,
              "borderTopRightRadius": 4,
            },
            Object {
              "lineHeight": 25,
              "paddingLeft": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.26)",
              "bottom": 0,
              "height": 2,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "transform": Array [
                Object {
                  "scaleY": 0.5,
                },
              ],
            }
          }
        />
        <View
          style={
            Object {
              "minHeight": 64,
              "paddingBottom": 0,
              "paddingTop": 0,
            }
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Enter your email"
            placeholderTextColor="rgba(0, 0, 0, 0.54)"
            rejectResponderTermination={true}
            selectionColor="#6200ee"
            style={
              Array [
                Object {
                  "flexGrow": 1,
                  "margin": 0,
                  "textAlign": "left",
                  "zIndex": 1,
                },
                Object {
                  "paddingLeft": 12,
                  "paddingRight": 12,
                },
                Object {
                  "height": 64,
                },
                Object {
                  "paddingBottom": 0,
                  "paddingTop": 0,
                },
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontSize": 20,
                  "fontWeight": undefined,
                  "textAlignVertical": "center",
                },
                Array [
                  Object {},
                ],
              ]
            }
            testID="emailInput"
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#6200ee",
          "borderColor": "transparent",
          "borderRadius": 25,
          "borderStyle": "solid",
          "borderWidth": 0,
          "elevation": 2,
          "justifyContent": "center",
          "marginVertical": 5,
          "minWidth": 64,
          "padding": 15,
          "shadowColor": "#000000",
          "shadowOffset": Object {
            "height": 0.75,
            "width": 0,
          },
          "shadowOpacity": 0.24,
          "shadowRadius": 1.5,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityState={
          Object {
            "disabled": undefined,
          }
        }
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "overflow": "hidden",
            },
            Object {
              "borderRadius": 25,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
              },
              undefined,
            ]
          }
        >
          <Text
            numberOfLines={1}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontFamily": "System",
                  "fontWeight": "400",
                  "textAlign": "left",
                },
                Array [
                  Object {
                    "letterSpacing": 1,
                    "marginHorizontal": 16,
                    "marginVertical": 9,
                    "textAlign": "center",
                  },
                  undefined,
                  Object {
                    "textTransform": "uppercase",
                  },
                  Object {
                    "color": "#ffffff",
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  Object {
                    "fontFamily": "System",
                    "fontWeight": "500",
                  },
                  undefined,
                ],
              ]
            }
          >
            <Text
              style={
                Object {
                  "fontSize": 18,
                  "fontWeight": "bold",
                  "textTransform": "uppercase",
                }
              }
            >
              Next
            </Text>
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
